packages/editor/src/components/provider/index.js

```diff
@@ -50,7 +50,25 @@ const fetchLinkSuggestions = async ( search, { perPage = 20 } = {} ) => {
 		} ),
 	} );
 
-	return map( posts, ( post ) => ( {
+	const categories = await apiFetch( {
+		path: addQueryArgs( '/wp/v2/search', {
+			search,
+			per_page: perPage,
+			type: 'category',
+		} ),
+	} );
+
+	const formats = await apiFetch( {
+		path: addQueryArgs( '/wp/v2/search', {
+			search,
+			per_page: perPage,
+			type: 'post-format',
+		} ),
+	} );
+
+	const results = concat( posts, categories, formats );
```

Comment:
Maybe sort them together? Otherwise categories will always come after posts, and formats after categories.

Reply:
sort by what? we it is "rarer" to link to index pages nowadays :) so they may well be last?

Reply:
I don't have a strong preference here, just bringing it up - I trust your judgement :-)